dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -965,16 +1014,30 @@ Spdp::data_received(const DataSubmessage& data,
     return;
   }
 
-  handle_participant_data(
-    (data.inlineQos.length() && disposed(data.inlineQos)) ?
-      DCPS::DISPOSE_INSTANCE : DCPS::SAMPLE_DATA,
-    pdata, to_opendds_seqnum(data.writerSN), from, false);
-
+  const bool from_relay = from == config_->spdp_rtps_relay_address();
 #ifdef OPENDDS_SECURITY
+  if (!from_relay && !ip_in_locator_list(from, pdata.participantProxy.metatrafficUnicastLocatorList) && !ip_in_AgentInfo(from, plist)) {
+    if (DCPS::DCPS_debug_level) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) Spdp::data_received - dropped IP: %C\n"), DCPS::LogAddr(from).c_str()));
+    }
+    return;
+  }
   if (!is_security_enabled()) {
     process_participant_ice(plist, pdata, guid);
   }
+#else
```

Comment:
```suggestion
#elif !defined OPENDDS_SAFETY_PROFILE
```

Reply:
Should we also put #ifndef OPENDDS_SAFETY_PROFILE in the #ifdef OPENDDS_SECURITY branch?